### PR TITLE
Phi-1.5 model bringup

### PR DESCRIPTION
**Pull request by velonica0**
### Ticket
Closes #19415

### Problem description
`microsoft/phi-1.5` model bringup on Tenstorrent Hardware (N300/N150)

### What's changed

- Added support for `bias` in each module of `tt_transformers` in Transformers layer. (MLP+Norm+lm_head)
- Add support for `two-layer MLP` in `tt_transformers`.
- Add support for `LayerNorm` in `tt_transformers`
- The Phi-1 model architecture lacks a Norm compared to Llama. Therefore, `tt_transformers` adds support for the `missing Norm`.
- Add support for `Partial Rotary Embedding`  in `tt_transformers` (ttnn.chunk can only be split evenly, not according to the specified ratio, see #25619)
- Given the conflict between o_proj (in Llama) and dense (in Phi-1.5), the call to the `map_meta_to_hf_keys `function was removed in the test, and the state_dict_ref was passed directly to the reference_model.
- phi-1.5 use MHA and MLP layers in parallel configuration, so change it. (see phi-1 paper)

### Notably
The possible reason for the low accuracy of phi-1.5 is

- `tt_transformers` and `huggingface` use different tensor formats when performing attention operations.(see #25619)
- The phi-1.5 model does not have the `chat_template` option and cannot chat.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-quick-trigger.yaml) CI passes (if applicable)
- [ ] [TG demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by velonica0** [comment]
@skhorasganiTT
"Could you please review this code? Thank you for your help."

**Comment by skhorasganiTT** [comment]
> @skhorasganiTT "Could you please review this code? Thank you for your help."

Hi @velonica0, my approval is not required for these changes. @mtairum or @gwangTT can take a look when they have time.

**Comment by velonica0** [comment]
Hi, @mtairum @gwangTT
Could you review my code?

**Comment by gwangTT** [comment]
Sorry for the delay in responding.

There is additional steps before we can merge this in:

    make a mirror branch that we can kick off CI runs from
    get successful CI runs to verify the accuracy and performance (or plan-b: I can run local tests to verify)
    pull from main and try to merge!

I can help you through the above steps. Thanks for contributing!

I see you have another PR on Phi-4, where I left the same comment as above. I think we should focus on getting one of the models (Phi-1.5 and Phi-4) merged first. I am leaning towards this one. Does that sound ok to you, @velonica0?

**Comment by gwangTT** [comment]
mirror branch created: mirror/velonica0/phi-1.5

**Comment by gwangTT** [comment]
Let's use the mirror branch to make further progress on this model, @velonica0.

This previous bounty model shows an example of what else needs to be done in the mirror branch: https://github.com/tenstorrent/tt-metal/pull/27508

**Comment by marty1885** [comment]
@velonica0 any luck keep updating the PR?

**Comment by velonica0** [comment]
> @velonica0 any luck keep updating the PR?

Ok, sorry I've been busy with other tasks recently, I will process this PR as soon as possible.

**Comment by velonica0** [comment]
> @velonica0 any luck keep updating the PR?

Could you please run the CI check to see if it passes? Thanks.
